3주차/[Lv.3]_야근_지수/조은서.js

```diff
@@ -0,0 +1,18 @@
+function solution(n, works) {
+  if (works.reduce((acc, cur) => acc + cur, 0) < n) return 0;
+  works.sort((a, b) => b - a);
+
+  while (n > 0) {
+    let i = 0;
+    works[i] -= 1;
+    while (works[i] <= works[i + 1]) {
+      i++;
+    }
+    const tmp = works[i];
+    works[i] = works[0];
+    works[0] = tmp;
```

Comment:
```suggestion
    [works[0], works[i]] = [works[i], works[0]]
```
이런식으로 정리하는게 더 가독성이 좋고 깔끔할 것 같습니다. (a.k.a JS 꼼수) 🙋🏻‍♂️

Reply:
저 원석님 코드를 보면서 이 부분 궁금했었는데 어떤 원리로 저 방식이 가능한가요??😲